### PR TITLE
WIP: Fix out of bounds slice issue on TerraformCliArgs

### LINKED ISSUE #411: Bug in CLI arg parsing
If you run `terragrunt force-unlock` (with no ID), you get an exception:

```
[terragrunt] 2018/02/01 16:17:46 runtime.errorString runtime error: slice bounds out of range
/private/tmp/terragrunt-20180130-90428-105chku/terragrunt-0.14.0/src/github.com/gruntwork-io/terragrunt/errors/errors.go:72 (0x10ed5b3)
/usr/local/Cellar/go/1.9.3/libexec/src/runtime/asm_amd64.s:509 (0x105724b)
        call32: CALLFN(·call32, 32)
/usr/local/Cellar/go/1.9.3/libexec/src/runtime/panic.go:491 (0x102b6d3)
        gopanic: reflectcall(nil, unsafe.Pointer(d.fn), deferArgs(d), uint32(d.siz), uint32(d.siz))
/usr/local/Cellar/go/1.9.3/libexec/src/runtime/panic.go:35 (0x102a41e)
        panicslice: panic(sliceError)
/private/tmp/terragrunt-20180130-90428-105chku/terragrunt-0.14.0/src/github.com/gruntwork-io/terragrunt/options/options.go:187 (0x112c9bc)
/private/tmp/terragrunt-20180130-90428-105chku/terragrunt-0.14.0/src/github.com/gruntwork-io/terragrunt/cli/cli_app.go:252 (0x14f7bee)
/private/tmp/terragrunt-20180130-90428-105chku/terragrunt-0.14.0/src/github.com/gruntwork-io/terragrunt/cli/cli_app.go:223 (0x14f761d)
/private/tmp/terragrunt-20180130-90428-105chku/terragrunt-0.14.0/src/github.com/gruntwork-io/terragrunt/cli/cli_app.go:196 (0x14f7540)
/private/tmp/terragrunt-20180130-90428-105chku/terragrunt-0.14.0/src/github.com/gruntwork-io/terragrunt/cli/cli_app.go:177 (0x14f720e)
/private/tmp/terragrunt-20180130-90428-105chku/terragrunt-0.14.0/src/github.com/gruntwork-io/terragrunt/vendor/github.com/urfave/cli/app.go:502 (0x1498542)
/private/tmp/terragrunt-20180130-90428-105chku/terragrunt-0.14.0/src/github.com/gruntwork-io/terragrunt/vendor/github.com/urfave/cli/app.go:268 (0x1496413)
/private/tmp/terragrunt-20180130-90428-105chku/terragrunt-0.14.0/main.go:20 (0x14fcf74)
/usr/local/Cellar/go/1.9.3/libexec/src/runtime/proc.go:195 (0x102d4f6)
        main: fn()
/usr/local/Cellar/go/1.9.3/libexec/src/runtime/asm_amd64.s:2337 (0x1059a91)
        goexit: BYTE    $0x90   // NOP
```

Looks like we're missing a [bounds check here](https://github.com/gruntwork-io/terragrunt/blob/master/options/options.go#L187).

### PR DESCRIPTION
**Pull request by sarkis**
Fixes #411

I decided to not use math.Min as the math package deals in float64 so settled on a Min function in the util package.

### CODE REVIEW

**Comment by sarkis** [comment]
@brikis98 had to move this over to a new PR - hit a weird issue on the other PR I had opened - I closed it, but don't think I have the right to delete it.

Made sure all tests ran and passed locally, ready for a CR on this.

File: util/min_test.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	testCases := []struct {
+		x int
+		y int
+		expected int
+	}{
+		{1, 2, 1},
+		{100, 4, 4},
+		{0, 25, 0},
+		{-1, 1, -1},
+		{0, -1, -1},
```
Comment (brikis98) [inline util/min_test.go]: Should we handle the corner cases mentioned here? https://golang.org/pkg/math/#Min. Also, check two equal numbers.

File: test/integration_test.go [hunk 2]
```diff
@@ -469,6 +469,91 @@ func TestPriorityOrderOfArgument(t *testing.T) {
...
+			"terraform force-unlock foo",
+		},
+		{
+			[]string{"force-unlock", "foo", "bar", "baz"},
+			"terraform force-unlock foo bar baz",
+		},
+		{
+			[]string{"force-unlock", "foo", "bar", "baz", "foobar"},
+			"terraform force-unlock foo bar baz foobar",
+		},
```
Comment (brikis98) [inline test/integration_test.go]: Nice tests!

File: util/min_test.go (hunk 1 above)
Comment (sarkis) [reply util/min_test.go]: Good catch on the 2 equal number check - ty! Added and feels like that made the test better :)

re: corner cases - I looked over those just now.. it seems most of them are only issues when working with `float64` types... I can't think of a case we'd get something like `-Inf` or `-0` since we are only going to be working with `int`s.

File: test/integration_test.go (hunk 2 above)
Comment (sarkis) [reply test/integration_test.go]: Thanks... so I hit reboot on the attempt writing tests first - what a difference it was... lesson learned.

**Comment by brikis98** [comment]
Haha, tests passed, but the `fmt` check failed. I just ran `make fmt` and committed to kick off the build again :)

**Comment by sarkis** [comment]
doh! I had to turn off my autoCommand for go fmt, it was trying to reformat code I hadn't touched - forget if it was on this project - but turning that back on :+1:

**Comment by brikis98** [comment]
New binaries should be here shortly: https://github.com/gruntwork-io/terragrunt/releases/tag/v0.14.1
